Core/Core/Views/ScannerViewController.swift

```diff
@@ -0,0 +1,217 @@
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        if (captureSession?.isRunning == false) {
+            captureSession.startRunning()
+        }
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        if (captureSession?.isRunning == true) {
+            captureSession.stopRunning()
+        }
+    }
+
+    func metadataOutput(_ output: AVCaptureMetadataOutput, didOutput metadataObjects: [AVMetadataObject], from connection: AVCaptureConnection) {
+        captureSession.stopRunning()
+
+        if let metadataObject = metadataObjects.first {
+            guard let readableObject = metadataObject as? AVMetadataMachineReadableCodeObject else { return }
+            guard let stringValue = readableObject.stringValue else { return }
+            AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
+            found(code: stringValue)
+        }
+    }
+
+    func found(code: String) {
+        delegate?.scanner(self, didScanCode: code)
+    }
+
+    override var prefersStatusBarHidden: Bool {
+        return true
+    }
+
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return .portrait
+    }
+
+    @objc func cancelTapped(_ sender: UIButton) {
+        dismiss(animated: true, completion: nil)
+    }
+}
+
+class QROverlayView: UIView {
+    let arc = CAShapeLayer()
+    var curveLength: CGFloat = 50
+
+    override var bounds: CGRect {
+        didSet { updateSize() }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        arc.lineWidth = 8
+        arc.fillColor = UIColor.clear.cgColor
+        arc.strokeColor = UIColor.named(.borderLightest).cgColor
+        arc.lineCap = .round
+        layer.addSublayer(arc)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func draw(_ rect: CGRect) {
+        let path = UIBezierPath()
+
+        // top left
+        path.move(to: CGPoint(x: 0, y: curveLength))
+        path.addQuadCurve(to: CGPoint(x: curveLength, y: 0), controlPoint: CGPoint(x: 0, y: 0))
+
+        // top right
+        path.move(to: CGPoint(x: rect.width - curveLength, y: 0))
+        path.addQuadCurve(to: CGPoint(x: rect.width, y: curveLength), controlPoint: CGPoint(x: rect.width, y: 0))
+
+        // bottom right
+        path.move(to: CGPoint(x: rect.width, y: rect.height - curveLength))
+        path.addQuadCurve(to: CGPoint(x: rect.width - curveLength, y: rect.height), controlPoint: CGPoint(x: rect.width, y: rect.height))
+
+        // bottom left
+        path.move(to: CGPoint(x: curveLength, y: rect.height))
+        path.addQuadCurve(to: CGPoint(x: 0, y: rect.height - curveLength), controlPoint: CGPoint(x: 0, y: rect.height))
+
+        arc.path = path.cgPath
+    }
+
+    func updateSize() {
+//        arc.frame = bounds
```

Comment:
do we still need this func? Or did implementing draw make this unnecessary?

Reply:
Ah left this here by mistake. Thanks!